### ISSUE TITLE
Add Support for IAsyncDisposable to Dispose of Tests

### ISSUE THREAD

**Issue by RehanSaeed** [xunit/xunit#2017]
`netcoreapp3.0` or `netstandard3.0` they added `IAsyncDisposable`. It would be nice if xUnit supported asynchronous disposal of test classes and called `DisposeAsync`, just like it supports calling `Dispose` today for `IDisposable`.

```c#
public class FooTest : IAsyncDisposable
{
    [Fact]
    public void Foo_When_Then()
    {
        // ...
    }

    public ValueTask DisposeAsync()
    {
        // ...
    }
}
```

**Comment by bradwilson** [comment]
This is already supported today through IAsyncLifetime, for test classes and fixtures.

**Comment by RehanSaeed** [comment]
I think it's also useful to use the built-in types which will eventually become the defacto standard for disposing of objects asynchronously. It's also more discoverable.

**Comment by bradwilson** [comment]
I understand. I'm offering you a mechanism that already works today so you don't have to wait.

**Comment by HakamFostok** [comment]
If I understood correctly, this is implemented now in xunit, right? <br/>
I have the most recent version of xunit and .net 6 but still xunit does not call `DisposeAsync` <br/>
I think this means that I have something wrong with my code, but xunit support this now, Correct?

**Comment by bradwilson** [comment]
This is implemented in v3, not v2. The v3 builds are still alpha-level builds not intended for production usage.

**Comment by mshwf** [comment]
> This is already supported today through IAsyncLifetime, for test classes and fixtures.

This means I should implement `InitializeAsync()`, but I only need `DisposeAsync`

**Comment by bartelink** [comment]
>This means I should implement InitializeAsync() as well, but I only need DisposeAsync

On V2, stuff is baked - there's little chance of anything changing (it's been noted elsewhere that there's a very high bar for anything). On V3 this does not apply. (Feel free to restate your comment if you're making one beside the fact that the outgoing construct was noisy for you?)

**Comment by mshwf** [comment]
I'm on V2 so I think will update the code once V3 is out.

**Comment by bradwilson** [comment]
Yes, correct, an empty `InitializeAsync` is needed in v2, and can be removed in v3 once you shift from `IAsyncLifetime` to `IAsyncDisposable`. (In v3, `IAsyncLifetime` in fact "inherits" from `IAsyncDisposable`, as you'd expect when moving forward technology-wise, even though it's a breaking change since the return types change from `Task` to `ValueTask`.)
